From 45e27e29894e50d6777f60f0bf6ede0ec0fa4c46 Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Tue, 10 Dec 2024 17:04:45 +0200
Subject: [PATCH] [SW-210370] preboot fit: integrate fit utils file

Integrated the fit utils and removed the redundant function
from existing code.

Removed the FIT_PROP macro's and use from hl_fit_if.h

Change-Id: I60fe8d02c4cc0f5bcce8e5db874dde56b93a4fcf
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index 879a396..0db71a2 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -64,29 +64,7 @@
 void do_yield(void) {}
 #endif
 
-#define FIT_IMAGES_PATH		"/images"
-#define FIT_CONFS_PATH		"/configurations"
-
-#define FIT_DESC_PROP		"description"
-#define FIT_LOADABLE_PROP	"loadables"
 #define FIT_HASHED_NODES_PROP	"hashed-nodes"
-#define FIT_DATA_PROP		"data"
-#define FIT_HASH_PROP		"hash"
-#define FIT_LOADADDR_PROP	"loadaddr"
-#define FIT_STARTADDR_PROP	"startaddr"
-#define FIT_ARCH_PROP		"arch"
-#define FIT_COMPRESSION_PROP	"compression"
-
-#define FIT_SEC_VERSION		"hl-sec-version"
-#define FIT_SEC_KEY_IDX		"hl-sec-key-idx"
-#define FIT_SIG_NODENAME	"signature"
-#define FIT_VALUE_PROP		"value"
-#define FIT_ALGO_PROP		"algo"
-#define FIT_KEY_NAME_HINT	"key-name-hint"
-#define FIT_PUB_KEY_PROP	"pub-key"
-#define FIT_PUB_KEY_EXT_PROP	"pub-key-ext"
-#define FIT_LOAD_TYPE_PROP	"load-type"
-#define FIT_PUB_KEY_HASH_PROP	"hash"
 
 #define FIT_MAX_PATH		200
 #define MAX_FDT_REGIONS		20
@@ -211,134 +189,6 @@
 	return false;
 }
 
-static const char *fit_loadables_name(const void *fit, int noffset, int id)
-{
-	return fdt_stringlist_get(fit, noffset, FIT_LOADABLE_PROP, id, NULL);
-}
-
-static int fit_find_config_node(const void *fit)
-{
-	int conf, node;
-	int dflt_conf_node = -1;
-
-	conf = fdt_path_offset(fit, FIT_CONFS_PATH);
-	if (conf < 0) {
-		EMB_LOG_ERR("Cannot find configuration node");
-		return -1;
-	}
-
-	for (node = fdt_first_subnode(fit, conf); node >= 0; node = fdt_next_subnode(fit, node)) {
-		const char *node_name = fdt_get_name(fit, node, NULL);
-
-		if (strcmp(FIT_CONF_NAME, node_name) == 0) {
-			dflt_conf_node = node;
-			break;
-		}
-	}
-
-	return dflt_conf_node;
-}
-
-int fit_foreach_property(const void *fit, const char *node_path,
-			fit_property_match_cb match_cb, void *data)
-{
-	int ret, count = 0;
-	int noffset, property;
-	const struct fdt_property *prop;
-
-	noffset = fdt_path_offset(fit, node_path);
-	if (noffset < 0) {
-		EMB_LOG_RAW_ERR("Cannot find entry ");
-		EMB_LOG_RAW(node_path);
-		EMB_LOG_RAW("\n\r");
-		return -1;
-	}
-
-	fdt_for_each_property_offset(property, fit, noffset) {
-		prop = fdt_get_property_by_offset(fit, property, NULL);
-		if (!prop)
-			continue;
-
-		ret = match_cb(fit, prop, data);
-		if (!ret)
-			count++;
-	}
-
-	return count;
-}
-
-int match_prop_version(const void *fit, const struct fdt_property *prop, void *data)
-{
-	const char *cmpstr = (const char *)data;
-	const char *prop_name = NULL;
-
-	prop_name = fdt_get_string(fit, fdt32_ld(&prop->nameoff), NULL);
-	if (strstr(prop_name, cmpstr)) {
-		EMB_LOG_RAW_INF("[FIT] ");
-		EMB_LOG_RAW(prop_name);
-		EMB_LOG_RAW(": ");
-		EMB_LOG_RAW(prop->data);
-		EMB_LOG_RAW("\n\r");
-		return 0;
-	}
-
-	return -1;
-}
-
-void fit_print_versions(void *fit)
-{
-	int ret;
-
-	ret = fit_foreach_property(fit, "/", match_prop_version, "version");
-	if (ret <= 0)
-		EMB_LOG_WRN("[FIT] No version properties were found.");
-}
-
-static int fit_loadables_node_get(const void *fit, int id)
-{
-	int images, conf = fit_find_config_node(fit);
-	const char *uname;
-
-	if (conf == -1) {
-		EMB_LOG_ERR("Wrong FIT type: Cannot find fit conf");
-		return conf;
-	}
-
-	uname = fit_loadables_name(fit, conf, id);
-
-	images = fdt_path_offset(fit, FIT_IMAGES_PATH);
-	return fdt_subnode_offset(fit, images, uname);
-}
-
-int fit_check_format(const void *fit)
-{
-	if (fdt_check_header(fit))
-		return FIT_FORMAT_HEADER_CHECK_FAILED;
-
-	/* mandatory / node 'description' property */
-	if (fdt_getprop(fit, 0, FIT_DESC_PROP, NULL) == NULL)
-		return FIT_FORMAT_NO_DESCRIPTION;
-
-	/* mandatory subimages parent '/images' node */
-	if (fdt_path_offset(fit, FIT_IMAGES_PATH) < 0)
-		return FIT_FORMAT_NO_IMAGES_PARENT_NODE;
-
-	return FIT_FORMAT_SUCCESS;
-}
-
-static int fit_get_int(const void *fit, int node, const char *prop_name,
-		       int default_val)
-{
-	const int *cell;
-	int len;
-
-	cell = fdt_getprop(fit, node, prop_name, &len);
-	if (cell && len >= sizeof(int))
-		return fdt32_to_cpu(cell[0]);
-
-	return default_val;
-}
-
 static int get_pub_key_node(const void *pub_key_fit, const char *key_name)
 {
 	int sig_node;
@@ -356,7 +206,7 @@
 	if (key_node_name[FIT_MAX_KEY_NAME - 1] != '\0')
 		return -1;
 
-	sig_node = fdt_subnode_offset(pub_key_fit, 0, FIT_SIG_NODENAME);
+	sig_node = fdt_subnode_offset(pub_key_fit, 0, HL_FIT_SIGNATURE_NODENAME);
 
 	return fdt_subnode_offset(pub_key_fit, sig_node, key_node_name);
 }
@@ -399,62 +249,6 @@
 
 __weak void preboot_security_boot_info_write_key_idx(u32 *key_idx) {}
 
-static int fit_getprop_u64(const void *fit, int node, const char *name,
-			   u64 *val64)
-{
-	const u32 *val32;
-	int len;
-
-	val32 = fdt_getprop(fit, node, name, &len);
-	if (!val32 || len != sizeof(u64))
-		return 1;
-
-	*val64 = ((u64)fdt32_to_cpu(val32[0])) << 32;
-	*val64 |= fdt32_to_cpu(val32[1]);
-	return 0;
-}
-
-int fit_init_resp(void *fit, size_t sz)
-{
-	int err;
-
-	err = fdt_create(fit, sz);
-	if (err)
-		return err;
-
-	err = fdt_finish_reservemap(fit);
-	if (err)
-		return err;
-
-	err = fdt_begin_node(fit, "");
-	if (err)
-		return err;
-
-	return 0;
-}
-
-int fit_fin_resp(void *fit, unsigned int *res_size)
-{
-	int err;
-
-	if (!fdt_size_dt_strings(fit)) {
-		*res_size = 0;
-		return 0; /* Empty response */
-	}
-
-	err = fdt_end_node(fit);
-	if (err)
-		return err;
-
-	err = fdt_finish(fit);
-	if (err)
-		return err;
-
-	*res_size = fdt_totalsize(fit);
-
-	return err;
-}
-
 __weak int fit_process_core_id(enum fit_core_id core_id, uint64_t loadaddr,
 				uint8_t *config)
 {
@@ -474,42 +268,13 @@
 	return preboot_decompress_image(dst, buf, len, arch, cmpr);
 }
 
-static int fit_load_get_type(const void *fit, int conf, u32 *type_out)
-{
-	const u32 *cell;
-	u32 type;
-	int len;
-
-	cell = fdt_getprop(fit, conf, FIT_LOAD_TYPE_PROP, &len);
-
-	/* If load type does not exists, assume it's u-boot */
-	if (!cell) {
-		*type_out = FIT_LOAD_TYPE_RUN;
-		return 0;
-	}
-
-	if (len != sizeof(*cell)) {
-		EMB_LOG_ERR("Corrupted load type");
-		return 1;
-	}
-
-	type = fdt32_to_cpu(*cell);
-	if (type >= FIT_LOAD_TYPE_COUNT) {
-		EMB_LOG_HEX_ERR("Unsupported load type ", type);
-		return 1;
-	}
-
-	*type_out = type;
-
-	return 0;
-}
-
 static bool fit_load_is_runnable(const void *fit, int conf)
 {
 	int err;
 	u32 type;
+	struct fit_error fit_err;
 
-	err = fit_load_get_type(fit, conf, &type);
+	err = fit_get_load_type(fit, &type, &fit_err);
 	if (err)
 		return false;
 
@@ -520,8 +285,9 @@
 {
 	int err;
 	u32 type;
+	struct fit_error fit_err;
 
-	err = fit_load_get_type(fit, conf, &type);
+	err = fit_get_load_type(fit, &type, &fit_err);
 	if (err)
 		return true;
 
@@ -540,12 +306,12 @@
 	int hash;
 	const char *prop;
 
-	hash = fdt_subnode_offset(fit, node, FIT_HASH_PROP);
-	prop = (const char *)fdt_getprop(fit, hash, FIT_ALGO_PROP, NULL);
+	hash = fdt_subnode_offset(fit, node, HL_FIT_HASH_PROP);
+	prop = (const char *)fdt_getprop(fit, hash, HL_FIT_ALGO_PROP, NULL);
 	if (strcmp(prop, FIT_SHA))
 		return NULL;
 
-	return (const u8 *)fdt_getprop(fit, hash, FIT_VALUE_PROP, NULL);
+	return (const u8 *)fdt_getprop(fit, hash, HL_FIT_VALUE_PROP, NULL);
 }
 
 __weak void sha_csum(const unsigned char *input, unsigned int ilen,
@@ -624,7 +390,8 @@
 {
 	const char *loadable_name;
 
-	loadable_name = fit_loadables_name(fit, fit_find_config_node(fit), id);
+	loadable_name = fit_get_loadable_image_name_from_conf(fit,
+			fit_get_conf_node_offset(fit), id);
 	if (!loadable_name) {
 		EMB_LOG_HEX_ERR("Wrong FIT format: loadable name not found: ", id);
 		return false;
@@ -662,8 +429,8 @@
 
 static enum fit_load_result fit_copy_images(const void *fit)
 {
-	int node, len, image_len, core_id_len;
-	int conf, num_images, i;
+	int node, image_len, core_id_len;
+	int conf, num_images, i, err;
 	const u8 *buf;
 	u8 hash[SHA_SUM_LEN];
 	u64 load, start;
@@ -672,17 +439,20 @@
 	const uint32_t *core_id;
 	bool executable = true;
 	const char *img_name;
+	size_t len;
 	void *thread_desc = get_thread_desc();
 
 	/* Get number of images in preboot configuration */
-	conf = fit_find_config_node(fit);
+	err = fit_get_conf_offset_and_num_of_loadables(fit, &conf, &num_images);
+	if (err)
+		return FIT_LOAD_CORRUPTED;
+
 	if (conf < 0)
 		return FIT_LOAD_CORRUPTED;
 
 	if (!fit_load_is_runnable(fit, conf))
 		return FIT_LOAD_CORRUPTED;
 
-	num_images = fdt_stringlist_count(fit, conf, FIT_LOADABLE_PROP);
 	if (num_images > FIT_MAX_IMAGES) {
 		EMB_LOG_HEX_ERR("Images in fit exceeds max limit: ", FIT_MAX_IMAGES);
 		return FIT_LOAD_CORRUPTED;
@@ -693,26 +463,29 @@
 
 	for (i = 0; i < num_images; i++) {
 		send_keep_alive(thread_desc);
-		node = fit_loadables_node_get(fit, i);
+		node = fit_get_loadable_node_offset_by_index(fit, conf, i);
 		if (node < 0) {
 			EMB_LOG_HEX_ERR("Wrong FIT format: loadable image not found: ", i);
 			return FIT_LOAD_NOT_RUNNABLE;
 		}
 
-		if (fit_getprop_u64(fit, node, FIT_LOADADDR_PROP, &load)) {
+		if (fit_get_loadaddr_from_offset(fit, node, &load)) {
 			EMB_LOG_HEX_ERR("Wrong FIT format: loadaddr not found for image: ", i);
 			return FIT_LOAD_BAD_LOAD_ADDR;
 		}
 
 		/* if start property is not found, then mark image as non executable */
-		if (fit_getprop_u64(fit, node, FIT_STARTADDR_PROP, &start)) {
+		if (fit_get_startaddr_from_offset(fit, node, &start)) {
 			executable = false;
 			start = load;
 		} else {
 			executable = true;
 		}
 
-		buf = fdt_getprop(fit, node, FIT_DATA_PROP, &len);
+		err = fit_get_data_from_offset(fit, node, (const void **)&buf, &len);
+		if (err)
+			return FIT_LOAD_CORRUPTED;
+
 		if (!buf) {
 			EMB_LOG_HEX_ERR("Wrong FIT format: data empty for image: ", i);
 			return FIT_LOAD_CORRUPTED;
@@ -725,7 +498,7 @@
 		}
 
 		cmpr = (const char *)fdt_getprop(fit, node,
-						 FIT_COMPRESSION_PROP,
+						 HL_FIT_COMPRESSION_PROP,
 						 NULL);
 		if (!cmpr) {
 			/*
@@ -746,8 +519,7 @@
 			return FIT_LOAD_BAD_LOAD_ADDR;
 		}
 
-		images[i].arch = (const char *)fdt_getprop(fit, node,
-							FIT_ARCH_PROP, NULL);
+		images[i].arch = fit_get_property_string(fit, node, HL_FIT_ARCH_PROP, NULL);
 		if (!images[i].arch) {
 			EMB_LOG_HEX_ERR("Wrong FIT format: arch empty for image: ", i);
 			return FIT_LOAD_CORRUPTED;
@@ -791,7 +563,7 @@
 
 		do_yield();
 		/* For some devices more operations are applied when a certain img is loaded */
-		img_name = fit_loadables_name(fit, conf, i);
+		img_name = fit_get_loadable_image_name_from_conf(fit, conf, i);
 		preboot_dev_specific_img_post_operations(img_name, images[i].loadaddr, image_len);
 	}
 
@@ -906,6 +678,24 @@
 	case FIT_FORMAT_UNSUPPORTED_LOAD_TYPE:
 		EMB_LOG_ERR("Unsupported load type");
 		break;
+	case FIT_FORMAT_NO_LOAD_TYPE:
+		EMB_LOG_ERR("No load type present in the conf");
+		break;
+	case FIT_FORMAT_NO_LOAD_ADDRESS:
+		EMB_LOG_ERR("No load address present in the node");
+		break;
+	case FIT_FORMAT_NO_DATA:
+		EMB_LOG_ERR("No data present in the node");
+		break;
+	case FIT_FORMAT_NO_CONF_NODE:
+		EMB_LOG_ERR("No conf node present in the fit");
+		break;
+	case FIT_FORMAT_NO_PROPERTY:
+		EMB_LOG_ERR("Given property not found in the node");
+		break;
+	case FIT_FORMAT_BAD_SIZE:
+		EMB_LOG_ERR("Property doesn't contain data in proper size");
+		break;
 	default:
 		EMB_LOG_HEX_ERR("unrecognized error value for: fit_format_result (%d)",
 			(unsigned int)errtype);
@@ -944,11 +734,12 @@
 
 int fit_verify(void *fit, struct fit_error *err)
 {
-	int node, len;
+	int node, ret;
 	int conf, num_images, i;
 	const u8 *buf;
 	u8 hash[SHA_SUM_LEN];
 	size_t fit_size = fdt_totalsize(fit);
+	size_t len;
 
 	err->fmt = fit_check_format(fit);
 	if (err->fmt) {
@@ -971,10 +762,20 @@
 	}
 
 	/* Get number of images in preboot configuration */
-	conf = fit_find_config_node(fit);
-	if (conf < 0) {
+	ret = fit_get_conf_offset_and_num_of_loadables(fit, &conf, &num_images);
+	if (ret || (conf < 0)) {
 		err->load = FIT_LOAD_CORRUPTED;
 		err->sec = FIT_SEC_NONE;
+		return 1;
+	}
+
+	if (num_images > FIT_MAX_IMAGES) {
+		EMB_LOG_HEX_ERR("Images in fit exceeds max limit: ", FIT_MAX_IMAGES);
+		err->load = FIT_LOAD_CORRUPTED;
+		return 1;
+	} else if (num_images < 0) {
+		EMB_LOG_HEX_ERR("Cant read images from fit: ", -num_images);
+		err->load = FIT_LOAD_CORRUPTED;
 		return 1;
 	}
 
@@ -986,27 +787,16 @@
 		return 1;
 	}
 
-	num_images = fdt_stringlist_count(fit, conf, FIT_LOADABLE_PROP);
-	if (num_images > FIT_MAX_IMAGES) {
-		EMB_LOG_HEX_ERR("Images in fit exceeds max limit: ", FIT_MAX_IMAGES);
-		err->load = FIT_LOAD_CORRUPTED;
-		return 1;
-	} else if (num_images < 0) {
-		EMB_LOG_HEX_ERR("Cant read images from fit: ", -num_images);
-		err->load = FIT_LOAD_CORRUPTED;
-		return 1;
-	}
-
 	for (i = 0; i < num_images; i++) {
-		node = fit_loadables_node_get(fit, i);
+		node = fit_get_loadable_node_offset_by_index(fit, conf, i);
 		if (node < 0) {
 			EMB_LOG_HEX_ERR("Wrong FIT format: loadable image not found: ", i);
 			err->load = FIT_LOAD_NOT_RUNNABLE;
 			return 1;
 		}
 
-		buf = fdt_getprop(fit, node, FIT_DATA_PROP, &len);
-		if (!buf) {
+		ret = fit_get_data_from_offset(fit, node, (const void **)&buf, &len);
+		if (ret || (!buf)) {
 			EMB_LOG_HEX_ERR("Wrong FIT format: data empty for image: ", i);
 			err->load = FIT_LOAD_CORRUPTED;
 			return 1;
@@ -1027,11 +817,7 @@
 	memset(&images, 0x0, sizeof(images));
 	memset(&arc_info, 0x0, sizeof(arc_info));
 
-	err->fmt = fit_check_format(fit);
-	if (err->fmt)
-		err->load = FIT_LOAD_CORRUPTED;
-	else
-		err->load = fit_copy_images(fit);
+	err->load = fit_copy_images(fit);
 
 	return !fit_err_is_success(err);
 }
@@ -1102,135 +888,7 @@
 		preboot_start_image(startaddr, valid_arc_img, core_id);
 }
 
-__weak size_t preboot_get_img_sec_desc_size(void)
-{
-	return 0;
-}
-
-int fit_validate_size(const void *fit, int len)
-{
-	size_t img_size = ROUND_UP4(fdt_totalsize(fit));
-
-	if (fdt_check_header(fit)) {
-		EMB_LOG_ERR("Bad FIT header");
-		return 1;
-	}
-
-	/* taking the security footer into account */
-	img_size += preboot_get_img_sec_desc_size();
-
-	return img_size != len;
-}
-
-int fit_get_load_type(const void *fit, u32 *load_type, struct fit_error *err)
-{
-	const u32 *cell;
-	u32 type = FIT_LOAD_TYPE_RUN;
-	int len, conf;
-
-	err->fmt = fit_check_format(fit);
-	if (err->fmt) {
-		err->load = FIT_LOAD_CORRUPTED;
-		return 1;
-	}
-
-	conf = fit_find_config_node(fit);
-	if (conf >= 0) {
-		cell = fdt_getprop(fit, conf, FIT_LOAD_TYPE_PROP, &len);
-		if (!cell) {
-			EMB_LOG_ERR("No load type. Assuming old FIT version");
-			*load_type = type;
-			return 0;
-		}
-		if (len != sizeof(*cell)) {
-			err->fmt = FIT_FORMAT_CURRUPTED_LOAD_TYPE;
-			return 1;
-		}
-
-		type = fdt32_to_cpu(*cell);
-		if (type >= FIT_LOAD_TYPE_COUNT) {
-			err->fmt = FIT_FORMAT_UNSUPPORTED_LOAD_TYPE;
-			return 1;
-		}
-	}
-
-	*load_type = type;
-
-	return 0;
-}
-
-static int fit_image_get_data(const void *fit, int noffset,
-		       const void **data, size_t *size)
-{
-	int len;
-
-	*data = fdt_getprop(fit, noffset, FIT_DATA_PROP, &len);
-	if (*data == NULL) {
-		*size = 0;
-		return -1;
-	}
-
-	*size = len;
-	return 0;
-}
-
-int fit_image_get_node(const void *fit, const char *image_uname)
-{
-	int noffset, images_noffset;
-
-	images_noffset = fdt_path_offset(fit, FIT_IMAGES_PATH);
-	if (images_noffset < 0) {
-		EMB_LOG_ERR("Can't find images parent node");
-		return images_noffset;
-	}
-
-	noffset = fdt_subnode_offset(fit, images_noffset, image_uname);
-	if (noffset < 0)
-		EMB_LOG_ERR("Can't get node offset for image unit name");
-
-	return noffset;
-}
-
-static int fit_conf_get_prop_node_index(const void *fit, int noffset,
-				 const char *prop_name, int index)
-{
-	const char *uname;
-	int len;
-
-	/* get image unit name from configuration property */
-	uname = fdt_stringlist_get(fit, noffset, prop_name, index, &len);
-	if (uname == NULL)
-		return len;
-
-	return fit_image_get_node(fit, uname);
-}
-
-int fit_image_get_loadable_data(const void *fit, const void **data,
-				size_t size)
-{
-	size_t actual_size;
-	int err, node, conf;
-
-	conf = fit_find_config_node(fit);
-	if (conf < 0)
-		return 1;
-
-	node = fit_conf_get_prop_node_index(fit, conf, FIT_LOADABLE_PROP, 0);
-	if (node < 0)
-		return 1;
-
-	err = fit_image_get_data(fit, node, data, &actual_size);
-	if (err)
-		return err;
-
-	if (actual_size != size)
-		return 1;
-
-	return 0;
-}
-
 #define HL_SEC_REGIONS_CNT	2
-#define HL_SEC_KEY_NAME		"hl_rsa3072"
 
 int fit_verify_img_sig(const void *img, size_t img_size, const void *sec_desc,
 		       size_t sec_desc_size, const void *pub_key_fit,
diff --git a/lib/preboot/fit.h b/lib/preboot/fit.h
index 941d1cf..e47b720 100644
--- a/lib/preboot/fit.h
+++ b/lib/preboot/fit.h
@@ -6,33 +6,10 @@
 #ifndef _HL_FIT_H_
 #define _HL_FIT_H_
 
-#include <libfdt.h>
 #include <hl_fit_if.h>
+#include <fit_utils.h>
 
 #define HL_SEC_KEY_NAME		"hl_rsa3072"
-
-enum fit_load_result {
-	FIT_LOAD_SUCCESS = 0,
-	FIT_LOAD_CORRUPTED,
-	FIT_LOAD_BAD_LOAD_ADDR,
-	FIT_LOAD_NOT_RUNNABLE,
-	FIT_LOAD_SEC_FAILED,
-};
-
-enum fit_format_result {
-	FIT_FORMAT_SUCCESS = 0,
-	FIT_FORMAT_HEADER_CHECK_FAILED,
-	FIT_FORMAT_NO_DESCRIPTION,
-	FIT_FORMAT_NO_IMAGES_PARENT_NODE,
-	FIT_FORMAT_CURRUPTED_LOAD_TYPE,
-	FIT_FORMAT_UNSUPPORTED_LOAD_TYPE,
-};
-
-struct fit_error {
-	enum fit_load_result load;
-	enum fit_format_result fmt;
-	enum fit_sec_failure sec;
-};
 
 void fit_load_print_error(enum fit_load_result errtype);
 void fit_format_print_error(enum fit_format_result errtype);
@@ -41,22 +18,10 @@
 int fit_verify(void *fit, struct fit_error *err);
 int fit_load_images(void *fit, struct fit_error *err);
 void fit_run_image(void);
-int fit_validate_size(const void *fit, int len);
-int fit_get_load_type(const void *fit, u32 *load_type, struct fit_error *err);
-int fit_image_get_loadable_data(const void *fit, const void **data,
-				size_t size);
-int fit_init_resp(void *fit, size_t sz);
-int fit_fin_resp(void *fit, unsigned int *res_size);
 int fit_verify_img_sig(const void *img, size_t img_size, const void *sec_desc,
 		       size_t sec_desc_size, const void *pub_key_fit,
 		       const u8 *sig, size_t sig_len);
 
-typedef int (*fit_property_match_cb)(const void *fit, const struct fdt_property *prop, void *data);
-int fit_foreach_property(const void *fit, const char *node_path, fit_property_match_cb match_cb,
-						void *data);
-void fit_print_versions(void *fit);
-int fit_check_format(const void *fit);
-int fit_image_get_node(const void *fit, const char *image_uname);
 int fit_node_check_hash(const void *fit, int node, const u8 *buf,
 						int len, void *hash);
 void preboot_start_image(u64 startaddr, bool valid_arc_img,
@@ -65,6 +30,8 @@
 bool get_fit_no_ret(void);
 void fit_clr_state(void);
 int get_mgmt_core_id(void);
+int preboot_verify_img_sig(const void *fit, size_t img_size,
+			   enum fit_sec_failure *res, u32 *key_idx);
 
 
 #endif /* _HL_FIT_H_ */
diff --git a/lib/preboot/fit_load.c b/lib/preboot/fit_load.c
index a76f1d5..0bbc702 100644
--- a/lib/preboot/fit_load.c
+++ b/lib/preboot/fit_load.c
@@ -44,7 +44,7 @@
 	uint32_t mask;
 
 	/* get array of 8 masks */
-	err = fit_image_get_loadable_data(fit, &data, 8 * sizeof(uint32_t));
+	err = fit_get_loadable_image_data(fit, &data, 8 * sizeof(uint32_t));
 	if (!err) {
 		fit_mask = (uint32_t *)data;
 		module_id = get_module_id();
